lib/shopify_api/rest/request.ex

```diff
@@ -74,7 +119,13 @@ defmodule ShopifyAPI.REST.Request do
 
   @impl true
   def process_response_body(body) do
-    JSONSerializer.decode(body)
+    with {:ok, results_map} when is_map(results_map) <- JSONSerializer.decode(body),
+         [{_key, results}] <- Map.to_list(results_map) do
+      results
+    else
+      other_val ->
+        other_val
```

Comment:
This `else` block shouldn't be necessary - the `with` macro will automatically generate one.